### ISSUE TITLE
Missing Libraries on Windows 7

### ISSUE THREAD

**Issue by Project-Aurora** [ggez/ggez#586]
I compiled the "01_super_simple.rs" Expample, on Windows 7 proper, i.e. not on MinGW or Linux Subsystem.
Runing the resulting .exe in the target\debug Folder leads to an Error Message stating that "XINPUT1_4.dll" and "api-ms-win-eventing-classicprovider-l1-1-0.dll" are missing.
As far as I can tell, the "api-ms-win-eventing-classicprovider-l1-1-0.dll" should be included in [this](https://support.microsoft.com/en-ie/help/2999226/update-for-universal-c-runtime-in-windows) Windows Update, which is already installed on my Computer.
On [this](https://docs.microsoft.com/en-us/windows/desktop/xinput/xinput-versions) Page Microsoft "sort of" states that "XINPUT1_4.dll" was introduced with Windows 8 and hasn't been backported to Windows 7.
It seems to me that the Purpose of XINPUT1.4 overlaps with that of SDL2, is it possible to use the latter instead?
I already followed the instructions in [BuildingForEveryPlatform](https://github.com/ggez/ggez/blob/master/docs/BuildingForEveryPlatform.md) and created a Dependency to SDL2 in the Projects Cargo.toml, this doesn't seem to change the Behavior I described above.

Side Note:
The "01_super_simple.rs" Example tries to set the DrawMode of a Circle using "graphics::DrawMode::fill(),", this results in an Error when trying to compile.
I think this is supposed to be: "graphics::DrawMode::Fill,".

**Comment by icefoxen** [comment]
`gilrs`, the library ggez currently uses for gamepads, does not support Windows versions <8, it seems.  I've talked with some people about this before and it's kind of annoying and not necessarily hard to fix, but `gilrs` people seem reluctant to do it(?).  I don't have a Win7 machine to test on at the moment, unfortunately.  Can you try modifying the `ContextBuilder` code as described [here](https://github.com/ggez/ggez/issues/577#issuecomment-463435500) and see if it works?

@Lokathor Can you give me a reference on the gilrs xinput issue and your branch so I actually remember them?  ggez has gotten enough complaints from Windows 7 users that this is now on the Really Needs Fixing list.

**Comment by Project-Aurora** [comment]
I tried the modified `ContextBuilder` Code, the resulting Binary still tries to load the `XINPUT1_4.dll`.
The `api-ms-win-eventing-classicprovider-l1-1-0.dll` Part is probably an Issue with my Setup.
Just to see what would happen, I added the DLLs mentioned in the first Post to the same Directory as the exe I'm trying to run.
Now I get an Application Error with the Code `0xc000007b`.
I used [Dependency Walker](http://www.dependencywalker.com/) to look for other Librarys that can't be loaded and found a bunch more.
<details><summary>Missing DLLs</summary>

```
API-MS-WIN-CORE-COM-L1-1-1.DLL
API-MS-WIN-CORE-ERRORHANDLING-L1-1-1.DLL
API-MS-WIN-CORE-FILE-L1-2-1.DLL
API-MS-WIN-CORE-HEAP-L1-2-0.DLL
API-MS-WIN-CORE-IO-L1-1-1.DLL
API-MS-WIN-CORE-LIBRARYLOADER-L1-2-0.DLL
API-MS-WIN-CORE-PROCESSTHREADS-L1-1-2.DLL
API-MS-WIN-CORE-QUIRKS-L1-1-0.DLL
API-MS-WIN-CORE-RTLSUPPORT-L1-2-0.DLL
API-MS-WIN-CORE-SYSINFO-L1-2-1.DLL
API-MS-WIN-RTCORE-NTUSER-PRIVATE-L1-1-4.DLL
API-MS-WIN-CORE-WINRT-ERROR-L1-1-0.DLL
API-MS-WIN-CORE-WINRT-L1-1-0.DLL
API-MS-WIN-CORE-WINRT-ROBUFFER-L1-1-0.DLL
API-MS-WIN-CORE-WINRT-STRING-L1-1-0.DLL
DCOMP.DLL
IESHIMS.DLL
```

</details>

Running the pre-Built [v0.5.0-alpha.0 Release of Zeremoth](https://github.com/ozkriff/zemeroth/releases/tag/v0.5.0-alpha.0) works, no complaining about missing Librarys and the Game seems to function properly.
When I compile this Release myself, it doesn't say anything about Librarys but I get the same Application Error `0xc000007b`.

Both the pre-Built and my broken Version are missing the same Librarys. (according to Dependency Walker)
<details><summary>Missing DLLs</summary>

```
API-MS-WIN-CORE-WINRT-ERROR-L1-1-0.DLL
API-MS-WIN-CORE-WINRT-L1-1-0.DLL
API-MS-WIN-CORE-WINRT-ROBUFFER-L1-1-0.DLL
API-MS-WIN-CORE-WINRT-STRING-L1-1-0.DLL
DCOMP.DLL
IESHIMS.DLL
```

</details>

Not sure what's going on here, I simply use `cargo build` to Build.

**Comment by Lokathor** [comment]
https://gitlab.com/gilrs-project/gilrs/issues/60

**Comment by icefoxen** [comment]
Zemeroth 0.5.0 is using ggez 0.4, it appears, so it's still using SDL2 instead of `winit`+`gilrs`.  Hence, no `gilrs` crate causing problems.

Also, on the `DrawMode::fill()` vs `DrawMode::Fill` issue, see this issue: https://github.com/ggez/ggez/issues/546 .  `fill()` is correct for `master` and `Fill` is correct for `devel`.  Need to get a proper release out and go back to developing on `devel` to avoid exactly these issues.  :/

**Comment by kampffrosch94** [comment]
I can run ggez examples on win7 when i build with the current master of gilrs, but not with the latest release on crates.io (0.7.1).

**Comment by icefoxen** [comment]
According to https://gitlab.com/gilrs-project/gilrs/issues/60 this was done a couple weeks ago, hopefully they release soon.  anyone wanting that release can harass the gilrs people for it.  :-P

**Comment by icefoxen** [comment]
Resolved in #588, it's fixed in `gilrs` master.  Not much we can do for now besides get `gilrs` to make a release with their fixed version.
